plugins/base/src/test/kotlin/signatures/FunctionalTypeConstructorsSignatureTest.kt

```diff
@@ -108,6 +108,31 @@ class FunctionalTypeConstructorsSignatureTest : BaseAbstractTest() {
         }
     }
 
+    @Test
+    fun `kotlin syntactic sugar function with param name of generic and functional type`() {
+        val source = source("""
+                            | @Target(AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.TYPE)
+                            | @MustBeDocumented
+                            | annotation class Fancy
+                            |
+                            | fun <T> f(): (param1: T, param2: @Fancy ()->Unit) -> String "
+                            """.trimIndent())
+        val writerPlugin = TestOutputWriterPlugin()
+
+        testInline(
+            source, configuration, pluginOverrides = listOf(writerPlugin)
+        ) {
+            renderingStage = { _, _ ->
+                writerPlugin.writer.renderedContent("root/example/index.html").lastSignature().firstSignature().match(
```

Comment:
Is `.lastSignature().firstSignature()` intentional?

Reply:
Yeas, there are two signatures on the page

Reply:
Sorry, I did not notice the redundant `firstSignature()`)(